lib/connectors_async/job_runner.rb

```diff
@@ -37,16 +37,15 @@ def start_job(job:, connector_class:, secret_storage:, params:)
         job.update_status(ConnectorsShared::JobStatus::FINISHED)
         job.update_cursors(new_cursors)
 
-        log("Job #{job.id} has finished successfully")
+        log_with_thread_id("Job #{job.id} has finished successfully")
       rescue StandardError => e
         job.fail(e)
-        log("Job #{job.id} failed: #{e.message}")
+        log_with_thread_id("Job #{job.id} failed: #{e.message}")
```

Comment:
Not really related to this change, but we should use ExceptionTracking here to get the stack trace.